text/src/od.rs

```diff
@@ -0,0 +1,718 @@
+                                print!("* ");
+                                continue;
+                            }
+
+                            print!("{}", current);
+                            previously = current;
+                        }
+                    }
+                    'f' => {
+                        let mut previously = String::new();
+                        for chunk in chunks {
+                            let value = match chunk.len() {
+                                4 => f32::from_be_bytes([chunk[3], chunk[2], chunk[1], chunk[0]])
+                                    as f64,
+
+                                8 => f64::from_be_bytes([
+                                    chunk[7], chunk[6], chunk[5], chunk[4], chunk[3], chunk[2],
+                                    chunk[1], chunk[0],
+                                ]),
+
+                                _ => {
+                                    return Err(Box::new(Error::new(
+                                        ErrorKind::Other,
+                                        format!("invalid type string `f{}`", num_bytes),
+                                    )))
+                                }
+                            };
+                            let current = format!("{} ", value);
+                            if previously == current && !config.verbose {
+                                print!("* ");
+                                continue;
+                            }
+
+                            print!("{}", current);
+                            previously = current;
+                        }
+                    }
+                    _ => {
+                        let mut previously = String::new();
+                        for &byte in local_buf {
+                            let current = format!("{:03o} ", byte);
+                            if previously == current && !config.verbose {
+                                print!("* ");
+                                continue;
+                            }
+
+                            print!("{}", current);
+                            previously = current;
+                        }
+                    }
+                }
+
+                println!(); // Print a newline after each line of bytes.
+            }
+        }
+
+        offset += 16; // Move to the next line of bytes.
+    }
+
+    offset -= 16;
+    offset = buffer.len() - offset;
+    if let Some(base) = config.address_base {
+        match base {
+            'd' => print!("{:07} ", offset),
+            'o' => print!("{:07o} ", offset),
+            'x' => print!("{:07x} ", offset),
+            'n' => (),
+            _ => print!("{:07} ", offset),
+        }
+    } else {
+        print!("{:07} ", offset);
+    }
+    Ok(())
+}
+
+/// Get a mapping of byte values to their named character representations.
+fn get_named_chars() -> HashMap<u8, &'static str> {
```

Comment:
Could've been a compile-time look-up table. Just accept the byte and match over it. Why the need for a map?